### PR TITLE
Windows eventchannel enhancement

### PR DESCRIPTION
**Pull request by cristgl**
This PR solves the issue https://github.com/wazuh/wazuh/issues/905, adding a new Windows events processing to get the data in JSON format.

### CODE REVIEW

File: src/analysisd/decoders/winevtchannel.c [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+    os_calloc(1, sizeof(OSDecoderInfo), winevt_decoder);
+    winevt_decoder->id = getDecoderfromlist(WINEVT_MOD);
+    winevt_decoder->name = WINEVT_MOD;
+    winevt_decoder->type = OSSEC_RL;
+    winevt_decoder->fts = 0;
+
+    mdebug1("WinevtInit completed.");
+}
+
+char *replace_string(const char *string, const char *old, const char *new){
```
Comment (TJOSERAFAEL) [inline src/analysisd/decoders/winevtchannel.c]: Use wstr_replace() function instead.

File: src/analysisd/decoders/winevtchannel.c [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+                            break;
+                        }
+                    } else {
+                        mdebug1("Malformed 'Event' field");
+                        break;
+                    }
+                }
+            }
+
+            num = end_event - find_event;
```
Comment (TJOSERAFAEL) [inline src/analysisd/decoders/winevtchannel.c]: Add boundary check. Malicious string could copy gigabytes of data.

File: src/analysisd/decoders/winevtchannel.c [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+
+                    XML_NODE child_attr = NULL;
+                    child_attr = OS_GetElementsbyNode(&xml, child[j]);
+                    int p = 0;
+
+                    while (child_attr && child_attr[p]){
+
+                        if(child[j]->element && !strcmp(child[j]->element, "System") && child_attr[p]->element){
+
+                            if (!strcmp(child_attr[p]->element, "Provider")) {
```
Comment (TJOSERAFAEL) [inline src/analysisd/decoders/winevtchannel.c]: Use **strncmp** for safer operation.

File: src/analysisd/decoders/winevtchannel.c [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+                            break;
+                        }
+                    } else {
+                        mdebug1("Malformed 'Message' field");
+                        break;
+                    }
+                }
+            }
+
+            num = end_msg - find_msg;
```
Comment (TJOSERAFAEL) [inline src/analysisd/decoders/winevtchannel.c]: Add boundary check.

File: src/logcollector/read_win_event_channel.c [hunk 5]
```diff
@@ -587,96 +485,70 @@ void send_channel_event(EVT_HANDLE evt, os_channel *channel)
...
+    
+    find_prov = strstr(xml_event, "Provider Name=");
+  
+    if(find_prov){
+        beg_prov = strchr(find_prov, '\'');
+        if(beg_prov){
+            end_prov = strchr(beg_prov+1, '\'');
+
+            if (end_prov){
+                num = end_prov - beg_prov - 1;
```
Comment (TJOSERAFAEL) [inline src/logcollector/read_win_event_channel.c]: Add boundary check.

**Comment by cristgl** [comment]
The channels added by default at the `ossec.conf` file are Security, System and Application, nevertheless, the ruleset contain some more channels, but maybe adding these ones to the configuration file could cause flooding. The extra channels are:

- Microsoft-Windows-Sysmon/Operational
- Microsoft Antimalware
- Microsoft-Windows-Windows Defender/Operational
- Microsoft-Windows-Eventlog

**Pull request merged**